op-service/txmgr/txmgr.go

```diff
@@ -111,14 +72,36 @@ type ETHBackend interface {
 // SimpleTxManager is a implementation of TxManager that performs linear fee
 // bumping of a tx until it confirms.
 type SimpleTxManager struct {
-	Config  // embed the config directly
+	cfg     Config // embed the config directly
 	name    string
 	chainID *big.Int
 
 	backend ETHBackend
 	l       log.Logger
 }
 
+// NewSimpleTxManager initializes a new SimpleTxManager with the passed Config.
+func NewSimpleTxManager(name string, l log.Logger, cfg Config) *SimpleTxManager {
+	if cfg.NumConfirmations == 0 {
+		panic("txmgr: NumConfirmations cannot be zero")
+	}
+	if cfg.NetworkTimeout == 0 {
+		cfg.NetworkTimeout = 2 * time.Second
+	}
+
+	return &SimpleTxManager{
+		chainID: cfg.ChainID,
+		name:    name,
+		cfg:     cfg,
+		backend: cfg.Backend,
+		l:       l.New("service", name),
```

Comment:
So this shouldn't be something like `"tx-mgr"`?

Reply:
Like `txmgr-for`?. This is to indicate which service is using the txmgr. We don't really need this when running the full system because we can sort logs by component. We kinda use this in op-e2e, but use a `role` to differentiate the different services: https://github.com/ethereum-optimism/optimism/blob/18b5f8ac35dd5a53e98a393c0ab1448c5b0d6267/op-e2e/setup.go#L148-L153

Reply:
Ah sorry, I didn't get that it's the key name 😅